### ISSUE TITLE
Response["state"] is inaccurate if polled instantly after switching entity state

### ISSUE THREAD

**Issue by oerkel47** [JarvyJ/HomeIntent#274]
I found that the response from the HA entity is inaccurate most of the time. A device that just got switched will report the old state if there is not a certain delay between switching and polling state. At least that's true for my setup and it's not super slow.

The obvious solution would be a delay (I tried 1 second and it worked). Delays suck, but 1 second wasn't too bad. The only other solution would be to poll the state beforehand and invert it for the response (in case of toggle). This would be a shame though, since it could produce incorrect values as well.

Maybe you have a better idea.

**Comment by JarvyJ** [comment]
Huh, interesting. I was worried about this when initially writing the code, but didn't experience it. My HA is behind an SSL proxy and over WiFi, which might have slowed the requests down just enough.

I wonder what the minimum delay would be. Would 1/2 or 1/4s work? I'll see if I can replicate it and test out some delays.

I might also look into if Home Assistant has a websocket that publishes changes or if there is an API attribute that indicates a change is in progress.

**Comment by oerkel47** [comment]
My HA is on a Pi4b on WiFi and the appliances I control work with MQTT. I guess the actual devices play a role as well (mine mostly run Tasmota). I haven't really tested different values yet, but I can do that if it helps. I know that an a[ttribute "assumed state" exists for states](https://www.home-assistant.io/docs/configuration/state_object/), maybe this is worth checking out.

**Comment by JarvyJ** [comment]
Can you check to see if it's just the MQTT devices and if they have an `assumed_state` in the attributes? You should be able to print the `response` to see what comes back.

I looked at some of my devices (unfortunately none of them are MQTT devices...) and they didn't have the `assumed_state` set even during transition.

I might take a closer look at the "last_updated" time attribute. But yeah, the quickest fix is definitely going to be putting in a bit of a delay. I'll continue looking at options.

**Comment by oerkel47** [comment]
Will do. I don't know if the assumed_state actually does anything useful for our case, I just thought the name looks interesting. I know that HA will check the actual state after changing it (if not set to optimistic mode) so it will only be updated internally if ithe device is able to provide the new state.If the device is slow this can still yield the incorrect result.

I think we have two options: use a delay that covers the response time of most devices but is not too long or go with an optimistic approach (which probably is the same as assumed state) as well.

**Comment by JarvyJ** [comment]
So I installed Home Intent on the same Pi 4 that is running Home Assistant and it _flies_. I was running it on a Pi 3 before and didn't realize how fast it all is. I now get the response state being incorrect.

For now, let's just keep it simple and add a delay. I'll add a `delay` parameter that gets passed in to the `get_entity` method in the API call, and then we can tweak the value across the board in one spot.

**Comment by JarvyJ** [comment]
Alright, the plot thickens. I went back and [read the documentation](https://developers.home-assistant.io/docs/api/rest) on the service call that makes the change. It is supposed to return _after_ completing the state change (or 10s timeout) and returns a list of all things that have changed while the service was being executed, and returns 
So I updated `_toggle_light`:
```python
  def _toggle_light(self, light):
      response = self.ha.api.call_service("light", "toggle", {"entity_id": light})
      print("1", response)
      response = self.ha.api.get_entity(light)
      print("2", response)
      response = self.ha.api.get_entity(light, delay=True)
      print("3", response)
      return response
```

and this is what it returned:
```
homeintent_1  | 1 []
homeintent_1  | 2 {'entity_id': 'light.kitchen', 'state': 'on', ...
homeintent_1  | delay
homeintent_1  | 3 {'entity_id': 'light.kitchen', 'state': 'off', ...
```
(it is supposed to be "off").

I did a similar thing to the `_toggle_switch` and these were the results:
```
homeintent_1  | 1 [{'entity_id': 'switch.fan', 'state': 'off', ...
homeintent_1  | 2 {'entity_id': 'switch.fan', 'state': 'off' ...
homeintent_1  | delay
homeintent_1  | 3 {'entity_id': 'switch.fan', 'state': 'off' ...
```
(the state is correctly "off")

The output from `_toggle_switch` is more with what I expect with a list of entities being returned. I think this might be a bug in Home Assistant.

**Comment by JarvyJ** [comment]
Okay, I also think this is from a newer release of Home Assistant, as I just updated yesterday to the latest.

I also tried it with a light that is not a hue_group (an individual bulb), and it works as expected:
```
homeintent_1  | 1 [{'entity_id': 'light.foyer_2', 'state': 'on' ...
homeintent_1  | 2 {'entity_id': 'light.foyer_2', 'state': 'on' ...
homeintent_1  | delay
homeintent_1  | 3 {'entity_id': 'light.foyer_2', 'state': 'on' ...
```

I suppose we could add the delay if the `call_service` comes back with an empty list. Also, I suppose there isn't a need to make the extra `get_entity` call at all if `call_service` comes back.
